agent/handlers/v1/response_test.go

```diff
@@ -0,0 +1,171 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+// http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package v1
+
+import (
+	"encoding/json"
+	"testing"
+	
+	apicontainer "github.com/aws/amazon-ecs-agent/agent/api/container"
+	apieni "github.com/aws/amazon-ecs-agent/agent/api/eni"
+	apitask "github.com/aws/amazon-ecs-agent/agent/api/task"
+	apitaskstatus "github.com/aws/amazon-ecs-agent/agent/api/task/status"
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	taskARN        = "t1"
+	family         = "sleep"
+	version        = "1"
+	containerID    = "cid"
+	containerName  = "sleepy"
+	eniIPv4Address = "10.0.0.2"
+)
+
+func TestTaskResponse(t *testing.T) {
+	expectedTaskResponseJSONString :=
```

Comment:
Instead of testing for string exactness, this should preferably test for the shape of the data: the expected keys and their values (so as to not test serialization/deserialization string format). This certainly handles the case at hand, but isn't very maintainable.

Reply:
It's worth noting that adding fields or reordering (through any cause) will make these fields unaligned and test fail.